joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/cpp/features17/Cpp17FeaturesTests.scala

```diff
@@ -131,14 +131,14 @@ class Cpp17FeaturesTests extends AstC2CpgSuite(fileSuffix = FileDefaults.CppExt)
           |}
           |""".stripMargin)
       cpg.method.nameNot("<global>").fullName.sorted shouldBe List(
-        "<lambda>0",
-        // TODO: fix scoping of nested lambdas
-        "<lambda>1",
-        "<lambda>2",
-        "<lambda>3",
-        "<lambda>4",
-        "addOne:int(int)",
-        "addOne:int(int).<lambda>5"
+        // TODO: fix return types of nested lambdas
+        "<lambda>0:ANY(int)",
```

Comment:
These names should be namespaced, they are clearly not unique in a full CPG. Since they are defined in the global scope, I guess using the file name is our only option.

It's not a regression though, so don't feel obligated to fix it in this PR.

Reply:
Hm, guess you are right. Filenames make sense, thats how we do in jssrc2cpg.